internal/dcache/rpc/server/handler.go

```diff
@@ -119,9 +115,15 @@ type mvInfo struct {
 	// Both StartSync and EndSync will quiesce IOs just before they move the mv into and out of syncing state, and
 	// resume IOs once the MV is safely moved into the new state.
 	//
-	// This boolean flag will be set by StartSync and EndSync to put the mv in quiesce state and it'll be read
-	// and honored by the various chunk related APIs - GetChunk, PutChunk, etc.
-	quiesceIOs atomic.Bool
+	// opMutex is used to ensure that only one operation, chunk IO (get, put or remove chunk) or
+	// sync (start sync or end sync) is in progress at a time.
+	// syncOpMutex is rw mutex. IO operations like get, put or remove chunk takes
+	// read lock on syncOpMutex, and sync operations like StartSync or EndSync takes write lock on it.
+	// opMutex is released immediately after acquiring the syncOpMutex.
+	// This ensures that the sync operation waits for the ongoing IO operations to complete.
+	// It also makes sure that no new IO operations will start as it has already acquired the opMutex.
```

Comment:
you will get the same behaviour just with RWMutex. Most RW locks take care of lot of readers starving occasional writer (as that's the usecase rwlocks are typically used for). I checked go implementation also does that

this is the comment from go explaining that

// If any goroutine calls [RWMutex.Lock] while the lock is already held by
// one or more readers, concurrent calls to [RWMutex.RLock] will block until
// the writer has acquired (and released) the lock, to ensure that
// the lock eventually becomes available to the writer.
// Note that this prohibits recursive read-locking.

This is also present in the auto-generated doc, which makes this an official behaviour that we can rely on

https://pkg.go.dev/sync#RWMutex

Reply:
Done, using only rw mutex